src/packages.jl

```diff
@@ -374,4 +387,53 @@ function locate_package(name::String)
   return String(Wrappers.ELM_REC(loaded, lname)[1])
 end
 
+
+##############################################################################
+#
+# special handling for the GAP package AtlasRep
+#
+# GAP.jl supports Julia artifacts for GAP packages.
+# One feature of the AtlasRep package is to download data files and
+# (if wanted) to store them in a local directory.
+# The path of this directory is either explicitly set in the user preference
+# `AtlasRepDataDirectory` of the package, or it is computed when the AtlasRep
+# package gets loaded.
+# The value of the user preference is an empty string if and only if
+# GAP does not know a path to a writable directory for the data files,
+# which means that downloaded files cannot be cached;
+# in this case, GAP.jl provides a scratchspace for the data,
+# and sets the value of the user preference accordingly.
+# We can use this scratchspace independent of the version of Julia, GAP.jl,
+# and AtlasRep.
+
+# The path to the scratchspace, will be filled by `init_atlasrep()`
+atlasrep_download_cache = ""
+
+# If GAP's AtlasRep package does not yet know a writable directory
+# for storing downloaded data, provide a Julia scratch space.
+function init_atlasrep()
+  atlasrep_download_cache == "" || return
+  atlasrep = GapObj("atlasrep")
+  AtlasRepDataDirectory = GapObj("AtlasRepDataDirectory")
+  val = Wrappers.UserPreference(atlasrep, AtlasRepDataDirectory)
+  if Wrappers.IsString(val)
+    # A default is set.
+    if length(val) == 0
```

Comment:
- The computed default for `AtlasRepDataDirectory` is a path inside the package directory *if this is writable*, and in this case I want to keep this default.
- Also in the case that the `gap.ini` file sets a path (such that no computed default gets set), I want to keep this path because the user wants it.
- If the path is *not* writable (which will happen if the artifact in question is read-only --currently it seems to be writable) then the computed default is an empty string (meaning that downloaded data are not cached), and then the idea is that GAP.jl can provide a scratchspace as a way to enable caching.

Accidentally storing the path to the scratchspace in a `gap.ini` file is an issue, good point.
More generally, we can override any value of `AtlasRepDataDirectory` that is not a valid (writable) directory path.

What I had mentioned already is that the currently proposed code ignores that a user may set an empty string in order to disable caching. One solution would be to check in the case of an empty string whether the package directory is writable, and if yes to conclude that the empty string is intentional, and to keep it.

Concerning the fact that the replacement of the default does not get triggered if one starts GAP without packages and later on loads GAP packages only on the GAP side, we could argue that the situation is not worse as it was before, and if one does not use the features from the Julia side than one does not get its advantages.

(I had thought whether a clean solution would be to put the relevant Julia code into the function for the default computation in AtlasRep's `PackageInfo.g` file. This would remove the problem that one might load all packages only with GAP's `LoadPackage`. The problem with the scratchspace path stored in `gap.ini` cannot be solved then, but it could be documented in AtlasRep that storing the computed value is not a good idea.)